rest/route/middleware.go

```diff
@@ -914,6 +916,46 @@ func (m *EventLogPermissionsMiddleware) ServeHTTP(rw http.ResponseWriter, r *htt
 	next(rw, r)
 }
 
+// NewGithubAuthMiddleware returns a middleware that verifies the payload.
+func NewGithubAuthMiddleware() gimlet.Middleware {
+	return &githubAuthMiddleware{}
+}
+
+type githubAuthMiddleware struct{}
+
+func (m *githubAuthMiddleware) ServeHTTP(rw http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+	githubSecret := []byte(evergreen.GetEnvironment().Settings().Api.GithubWebhookSecret)
+
+	payload, err := github.ValidatePayload(r, githubSecret)
+	if err != nil {
+		grip.Error(message.WrapError(err, message.Fields{
+			"source":  "GitHub hook",
+			"message": "rejecting GitHub webhook",
+			"msg_id":  r.Header.Get("X-Github-Delivery"),
+			"event":   r.Header.Get("X-Github-Event"),
+		}))
+		gimlet.WriteResponse(rw, gimlet.MakeJSONErrorResponder(errors.Wrap(err, "validating GitHub payload")))
+		return
+	}
+
+	r = setPayload(r, payload)
+	next(rw, r)
+}
+
+func setPayload(r *http.Request, payload []byte) *http.Request {
+	return r.WithContext(context.WithValue(r.Context(), payloadKey, payload))
+}
+
+func getPayload(ctx context.Context) []byte {
```

Comment:
Same thing as above - if this is something specific to just the GitHub webhook, I would define it as a method on `githubHookApi`.

Reply:
Made it more specific. I would avoid making it part of githubHookApi as that's used in the route handler, while these functions are used in both the middleware and route handler.